File: lab/python/pares/par_impar_ternario.py
def is_par_or_impar(num):
    """Retorna 'Par' se num é par, 'Impar' se num é ímpar, ou 'Não é inteiro'
    se num não é um inteiro."""
    if isinstance(num, bool):
        return "Não é inteiro"

    if not isinstance(num, (int, float)):
        return "Não é inteiro"
    return "Par" if num % 2 == 0 else "Impar"


# testes
def testes():
    for numeros in range(-5, 6):
        print(f"{numeros}: {is_par_or_impar(numeros)}")

    decimais = [-3.5, -2.0, 0.0, 4.2, 7.7, "salve", True, None]
    for n in decimais:
        print(f"{n}: {is_par_or_impar(n)}")


testes()

File: lab/javascript/pares/par_impar_ternario.js
function is_par_or_impar(num) {
  // Retorna 'Par' se num é par, 'Impar' se num é ímpar, ou 'Não é inteiro' caso contrário.
  if (typeof num === "boolean") {
    return "Não é inteiro";
  }

  if (typeof num !== "number" || !Number.isFinite(num)) {
    return "Não é inteiro";
  }

  if (!Number.isInteger(num)) {
    return "Não é inteiro";
  }

  return num % 2 === 0 ? "Par" : "Impar";
}

function testes() {
  // Testes com inteiros
  for (let i = -5; i <= 5; i++) {
    console.log(`${i}: ${is_par_or_impar(i)}`);
  }

  // Testes com decimais, strings e booleanos
  const decimais = [-3.5, -2.0, 0.0, 4.2, 7.7, "salve", true, null];
  for (let n of decimais) {
    console.log(`${n}: ${is_par_or_impar(n)}`);
  }
}

testes();
